### PR TITLE
Automatically flip the board if human plays the black side against an engine

### PR DESCRIPTION
**Pull request by alwey**
It is more convenient to play with an automatically flipped board. #148

### CODE REVIEW

**Comment by alwey** [comment]
please swap(the, plays) in title

File: projects/gui/src/mainwindow.cpp [hunk 1]
```diff
@@ -564,6 +564,16 @@ void MainWindow::setCurrentGame(const TabData& gameData)
 		m_evalWidgets[i]->setPlayer(player);
 	}
 
+	// if a human plays the black side against an engine then flip the board
+	if (m_players[0]->isHuman() != m_players[1]->isHuman())
```
Comment (ilaripih) [inline projects/gui/src/mainwindow.cpp]: I don't think it makes sense to loop over the players when we know that the white player is always `m_players[0]` or `m_players[Chess::Side::White]` and black player is `m_players[1]` or `m_players[Chess::Side::Black]`. I think this should be all that is needed:

```
if (!m_players[Chess::Side::White]->isHuman()
&&  m_players[Chess::Side::Black]->isHuman())
	m_gameViewer->boardScene()->flip();
```

Also, as a side note, taking a reference to a pointer type (`const auto& player`) doesn't usually make sense.

Comment (alwey) [inline projects/gui/src/mainwindow.cpp]: Thank you. The simplified code is much better.

**Comment by alwey** [comment]
Re-based and pushed new version.

**Pull request merged**